### PR TITLE
Log failed itin checks and bus notifs

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [na] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [na] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR adds logging to track data sent for trip monitoring to OTP and for bus notifications to respective services.

### CODE REVIEW

**Review by JymDyerIBI — APPROVED** [review]
👍  Good changes. I have something like the first one in my own repo (a `LOG.debug()` line) but I don't check those in.

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java [hunk 1]
```diff
@@ -301,7 +302,9 @@ private boolean makeOTPRequestAndUpdateMatchingItineraryInternal() {
         }
 
         // If this point is reached, a matching itinerary was not found
-        LOG.warn("No comparison itinerary found in otp response for trip");
+        LOG.warn("No comparison itinerary found in otp response for trip - params: {}", JsonUtils.toJson(this.trip.otp2QueryParams));
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java]: Nit. `this.` is not needed.

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java [hunk 2]
```diff
@@ -301,7 +302,9 @@ private boolean makeOTPRequestAndUpdateMatchingItineraryInternal() {
...
+        LOG.warn("No comparison itinerary found in otp response for trip - saved itinerary: {}", JsonUtils.toJson(this.trip.itinerary));
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java]: Nit. `this.` is not needed.

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java [hunk 3]
```diff
@@ -301,7 +302,9 @@ private boolean makeOTPRequestAndUpdateMatchingItineraryInternal() {
...
+        LOG.warn("No comparison itinerary found in otp response for trip - OTP itineraries: {}", JsonUtils.toJson(otpResponse.plan.itineraries));
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java]: Should this guard against a NPE in case plan is null?

**Review by br648 — COMMENTED** [review]
Is it also worth adding a warning at the top of the method in case the `otpResponse` is null?

**Comment by binh-dam-ibigroup** [comment]
I am also going to add logging for when an itinerary check fails on POSTing a monitored trip.

**Review by JymDyerIBI — APPROVED** [review]
Changes LGTM.

**Review by JymDyerIBI — APPROVED** [review]
Change approved.

**Review by JymDyerIBI — APPROVED** [review]
Yes, these make sense. (Looked it over, now will look at other PR.)

**Review by JymDyerIBI — APPROVED** [review]
👍  LGTM

File: src/main/java/org/opentripplanner/middleware/models/ItineraryExistence.java [hunk 4]
```diff
@@ -199,31 +206,43 @@ public void checkExistence(MonitoredTrip trip) {
...
+
+                // Handle response if valid itineraries exist.
+                if (plan != null && plan.itineraries != null) {
+                    for (Itinerary itineraryCandidate : plan.itineraries) {
+                        // If a matching itinerary on the same service day as the request date is found,
+                        // save the date with the matching itinerary.
+                        // (The matching itinerary will replace the original trip.itinerary.)
+                        if (
+                            ItineraryUtils.occursOnSameServiceDay(itineraryCandidate, otpRequest.dateTime, tripIsArriveBy) &&
+                                ItineraryUtils.itinerariesMatch(referenceItinerary, itineraryCandidate)
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/models/ItineraryExistence.java]: Nit. Reduce indentation.

File: src/main/java/org/opentripplanner/middleware/models/ItineraryExistence.java [hunk 5]
```diff
@@ -199,31 +206,43 @@ public void checkExistence(MonitoredTrip trip) {
...
+                            hasMatchingItinerary = true;
+                        }
+                    }
+                }
+
+                if (!hasMatchingItinerary) {
+                    // If no match was found for the date, mark day of week as non-existent for the itinerary.
+                    result.handleInvalidDate(otpRequest.dateTime);
+
+                    // Log if the itinerary didn't exist "today"
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/models/ItineraryExistence.java]: Nit. Full stop.

File: src/main/java/org/opentripplanner/middleware/models/ItineraryExistence.java [hunk 6]
```diff
@@ -199,31 +206,43 @@ public void checkExistence(MonitoredTrip trip) {
...
+                    if (index == 1) {
+                        LOG.warn("Itinerary existence check failed 'today' for trip {} - params: {}", trip.id, JsonUtils.toJson(trip.otp2QueryParams));
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/models/ItineraryExistence.java]: Nit. Could this be done in one warning? If not, could `Itinerary existence check failed 'today' for trip` be put into a separate String and reused?

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java [hunk 7]
```diff
@@ -301,7 +305,9 @@ private boolean makeOTPRequestAndUpdateMatchingItineraryInternal() {
         }
 
         // If this point is reached, a matching itinerary was not found
-        LOG.warn("No comparison itinerary found in otp response for trip");
+        LOG.warn("No comparison itinerary found for trip {} - params: {}", trip.id, JsonUtils.toJson(trip.otp2QueryParams));
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java]: Nit. Similar comment to above.

Comment (br648) [reply src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java]: Or perhap create a method to control the logging which is called from both places.

**Review by br648 — APPROVED** [review]
Approving with a few "nits"!

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java (hunk 7 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java]: @br648 I extracted the common code in ccbde41de3ab6bb5793214ed4293e5ae088affd6, let me know if https://github.com/ibi-group/otp-middleware/pull/263/commits/6f0de6e5232a247fea0a7a135e5b88f7b9271fe9 looks better than what was there before.

Comment (br648) [reply src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java]: @binh-dam-ibigroup Looks good now. Thanks.

**Pull request merged**